modules/instance_template/main.tf

```
@@ -99,6 +99,13 @@ resource "google_compute_instance_template" "tpl" {
     network            = var.network
     subnetwork         = var.subnetwork
     subnetwork_project = var.subnetwork_project
+    dynamic "access_config" {
+      for_each = var.public_ip ? [var.access_config] : []
```

Based on the requested variable change:

```suggestion
      for_each = var.access_config
```